File: BruteForce/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    # 조건
    # 1. width >= length
    # 2. 2 * width + 2 * length - 4 = brown
    # 3. (width - 2) * (length - 2) = yellow
    for i in range(1, total + 1): 
        if total % i == 0:
            length = i
            width = int(total / length)
            if width >= length and ((2 * width) + (2 * length)) - 4 == brown and (width - 2) + (length - 2) == yellow:
                print(length)
                print(width)
                return [width, length]
    
    return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.java
import java.util.*;
class Solution {
    static int width, height;
    
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        // 1. 노란 격자 부터 구하기
        int yellowWidth = yellow;
        int yellowHeight = 1;
        for (int i = 1; i <= yellow; i++) {
            if (yellow % i != 0) continue;
            
            int tempHeight = i;
            int tempWidth = yellow / tempHeight;

            // 카펫의 가로 길이는 세로 길이와 같거나, 세로 길이보다 깁니다.
            if (tempWidth >= tempHeight) {
                if ((tempWidth + 2) * (tempHeight + 2) != (brown + yellow)) continue;
                yellowWidth = tempWidth;
                yellowHeight = tempHeight;
            } 
        }
        System.out.println("최종 yellowWidth: " + yellowWidth + "yellowHeight: " + yellowHeight);
        
        answer[0] = yellowWidth + 2;
        answer[1] = yellowHeight + 2;
        
        return answer;
    }
}